make/migrations/postgresql/0015_1.10.0_schema.up.sql

```diff
@@ -0,0 +1,12 @@
+/** Add table for immutable tag  **/
+CREATE TABLE immutable_tag_rule
+(
+  id            SERIAL PRIMARY KEY NOT NULL,
+  project_id    int NOT NULL,
+  repo_filter  text,
+  tag_filter   text,
+  decoration    varchar(100) NOT NULL DEFAULT 'matches',
```

Comment:
the decoration is for repo or tag? each of them should have one `decoration`?

Or, we can use a JSON data to store all of the parameters?

Reply:
Remove decoration